src/groups/mqb/mqbblp/mqbblp_relayqueueengine.h

```diff
@@ -127,16 +127,368 @@ struct RelayQueueEngine_AppState : QueueEngineUtil_AppState {
 
     // CREATORS
     explicit RelayQueueEngine_AppState(
-        unsigned int                             upstreamSubQueueId,
-        const bsl::string&                       appId,
-        bslma::ManagedPtr<mqbi::StorageIterator> iterator,
-        mqbi::Queue*                             queue,
-        bdlmt::EventScheduler*                   scheduler,
-        const mqbu::StorageKey&                  appKey,
-        Routers::QueueRoutingContext&            queueContext,
-        bslma::Allocator*                        allocator = 0);
+        unsigned int                  upstreamSubQueueId,
+        const bsl::string&            appId,
+        mqbi::Queue*                  queue,
+        bdlmt::EventScheduler*        scheduler,
+        Routers::QueueRoutingContext& queueContext,
+        bslma::Allocator*             allocator = 0);
 };
 
+struct PushStream {
```

Comment:
This documentation comment is quite terse, and seems to put explaining why this data structure is shaped the way it is before what properties the data structure has.  I could eventually parse that information out, but a sentence at the start of "This data structure stores contexts of PUSHs, allowing for efficient sequential access by GUID and by App, and random-access removal for GUID."  All the information here is good, but I could only make sense of it since you've explained the design before.

Reply:
Drawing out the shape of the data structure also helped me here.  I'm not sure if ASCII art is truly worth it, but it's complex to see what this data structure is just from the code (even though it's not terribly complex at a high level).